x-pack/plugins/enterprise_search/server/lib/connectors/generate_connector_name.ts

```diff
@@ -16,24 +16,51 @@ import { indexOrAliasExists } from '../indices/exists_index';
 
 export const generateConnectorName = async (
   client: IScopedClusterClient,
-  connectorType: string
+  connectorType: string,
+  userConnectorName?: string
 ): Promise<{ apiKeyName: string; connectorName: string; indexName: string }> => {
   const prefix = toAlphanumeric(connectorType);
   if (!prefix || prefix.length === 0) {
-    throw new Error('Connector type is required');
+    throw new Error('Connector type or connectorName is required');
   }
-  for (let i = 0; i < 20; i++) {
-    const connectorName = `${prefix}-${uuidv4().split('-')[1]}`;
-    const indexName = `connector-${connectorName}`;
-
-    const result = await indexOrAliasExists(client, indexName);
-    if (!result) {
+  if (userConnectorName) {
+    let indexName = `connector-${userConnectorName}`;
+    const resultSameName = await indexOrAliasExists(client, indexName);
+    // index with same name doesn't exist
+    if (!resultSameName) {
       return {
-        apiKeyName: indexName,
-        connectorName,
+        apiKeyName: userConnectorName,
+        connectorName: userConnectorName,
         indexName,
       };
     }
+    // if the index name already exists, we will generate until it doesn't for 20 times
```

Comment:
all of this should be in a utility function that we unit test IMO

Reply:
You are right. I will try to check when I wake up tomorrow. It is quite late :D